tests/test_sequence_validation.py

```diff
@@ -0,0 +1,337 @@
+# This file is part of khmer, https://github.com/dib-lab/khmer/, and is
+# Copyright (C) 2013-2015, Michigan State University.
+# Copyright (C) 2015-2016, The Regents of the University of California.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#
+#     * Redistributions in binary form must reproduce the above
+#       copyright notice, this list of conditions and the following
+#       disclaimer in the documentation and/or other materials provided
+#       with the distribution.
+#
+#     * Neither the name of the Michigan State University nor the names
+#       of its contributors may be used to endorse or promote products
+#       derived from this software without specific prior written
+#       permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Contact: khmer-project@idyll.org
+# pylint: disable=missing-docstring,invalid-name
+
+# Tests for the ReadParser and Read classes.
+from __future__ import print_function
+from __future__ import absolute_import
+from khmer import _Countgraph, _Counttable, _SmallCountgraph, _SmallCounttable
+from khmer import _Nodegraph, _Nodetable
+from khmer import ReadParser, Nodegraph, _GraphLabels
+from . import khmer_tst_utils as utils
+import pytest
+
+
+PRIMES_1m = [1000003, 1009837]
+
+
+# all the table types!
+@pytest.fixture(params=[_Countgraph, _Counttable, _SmallCountgraph,
+                        _SmallCounttable, _Nodegraph, _Nodetable])
+def tabletype(request):
+    return request.param
+
+
+# all the counting types!
+@pytest.fixture(params=[_Countgraph, _Counttable, _SmallCountgraph,
+                        _SmallCounttable])
+def countingtype(request):
+    return request.param
+
+
+# all the graph types!
+@pytest.fixture(params=[_Countgraph, _Nodegraph])
+def graphtype(request):
+    return request.param
+
+
+@pytest.fixture
```

Comment:
Now that I've looked this up: can we change this back to `yield` and change the decorator to `@pytest.yield_fixture`? Link to the pytest 2.9 docs: http://doc.pytest.org/en/2.9.2/fixture.html#fixture-finalization-executing-teardown-code